src/activity/iActivity.hxx

```diff
@@ -9,11 +9,14 @@
 #include "../GameService.hxx"
 #include "../model/iModel.hxx"
 #include "../activity/ActivityType.hxx"
+#include "../controller/iController.hxx"
+#include "../services/MouseController.hxx"
 
 class iActivity : public GameService, public virtual iView
 {
   class Window & m_Window;
   std::vector<iModelPtr> m_States;
+  std::vector<iControllerPtr> m_Controllers;
```

Comment:
This is maybe personal style but I prefer to see full object type rather than alias. e.g. std::unique_ptr<iModel> rather than iModelPtr. It only hides important information that it is actually unique_ptr (which lead me first to the conclusion it creates memory leaks). I think the only few valid places exist. Specify type alias is the opaque type (can be changed in the future e.g. size_t) or when the real type is not known.

Reply:
I will create a seperate PR for all these changes